Added share functionality for Blogposts

VangelisDimi opened the pull request:
Added a share button at the bottom of every post.
It's a share button if the browser supports native share functionality and if not , it's just a copy link button.
If needed , I think we can add a shortened URL functionality for sharing pretty easily.

Closes #49

VangelisDimi committed:
share functionality

skourismanolis commented:
Good work, can you please add some icons for copy and share (@mgiannopoulos24 recommends https://icones.js.org/) 
and also we should change the text to be greek, "Αντιγραφή συνδέσμου" or something.

Can you also resolve the conflicts? You can do so by merging the target branch into your own.

mgiannopoulos24 commented:
You can choose any collection. Most of the icons have ready-to-copy Astro components.

VangelisDimi committed:
Merge branch 'astro' of https://github.com/kinimatografiko-pofpa/bobina-web into various-fixes

VangelisDimi committed:
add icons to share buttons and change text

VangelisDimi commented:
I changed the 'copy link' text to Greek and added icons to the share buttons.
<img width="521" height="227" alt="image" src="https://github.com/user-attachments/assets/a35f426a-5d43-424a-815b-094805111c68" />

mgiannopoulos24 commented:
Could you add a lightblue background on the share icon and with rounded borders?

VangelisDimi commented:
I tried a different design based on your idea. It looks like this.
<img width="292" height="148" alt="image" src="https://github.com/user-attachments/assets/95a28eba-8f5b-48c2-80da-eadb254ed9fc" />
Having the border only on the icon didn't look right. 
When hovered it looks like this.
<img width="288" height="97" alt="image" src="https://github.com/user-attachments/assets/d9b9d978-6438-4632-9da6-360246518a97" />
It looks nice but I fear that it might clash with the rest of the design.
@skourismanolis what do you think ? Should I keep it as it is above or use this design ? (The colors can be changed to something different).

> Could you add a lightblue background on the share icon and with rounded borders?

mgiannopoulos24 commented:
@VangelisDimi I have a better idea. Since the site relies on black, white and red you could do this.

The button by default should utilize `@apply bg-primary;` and when hovered it should utilize `@apply bg-primary_bold;`. Try this if you want. Blue just doesn't make sense, it was my fault suggesting it.

skourismanolis commented:
I agree that blue isn't in the design files, let's keep the red and when hovered just make it a bit darker. You can also experiment with adding a bit of padding to the button. In the end do what feels right to you @VangelisDimi

VangelisDimi committed:
change share buttons styling

VangelisDimi commented:
I ended up with this design.
<img width="265" height="120" alt="image" src="https://github.com/user-attachments/assets/5a088aa0-7651-4210-8552-2b6f9dc3babe" />
I didn't add any hover effects because I found them to be too much.
The copy link button becomes gray when pressed.
